Add aws authentication method documentation (#47922)
elastic/beats: docs/reference/filebeat/filebeat-input-httpjson.md

::::{note}
AWS auth settings are disabled if either `enabled` is set to `false` or the `auth.aws` section is missing.
::::

### `auth.aws.access_key_id` [_auth_aws_access_key_id]

```{applies_to}
stack: ga 9.3.0
```

The AWS access key id. It should be used alongside with [`auth.aws.secret_access_key`](#_auth_aws_secret_access_key).

### `auth.aws.secret_access_key` [_auth_aws_secret_access_key]

```{applies_to}
stack: ga 9.3.0
```

The AWS secret access key. It should be used alongside with [`auth.aws.access_key_id`](#_auth_aws_access_key_id).

::::{note}
Direct keys ([`auth.aws.access_key_id`](#_auth_aws_access_key_id) and [`auth.aws.secret_access_key`](#_auth_aws_secret_access_key)) can not be set alongside with shared credentials file ([`auth.aws.shared_credential_file`](#_auth_aws_shared_credential_file)) if both are set, the direct keys takes precedence.

If neither the direct keys nor the shared credentials file is set, AWS SDK [LoadDefaultConfig](https://pkg.go.dev/github.com/aws/aws-sdk-go-v2/config#LoadDefaultConfig) will be used.
::::

### `auth.aws.session_token` [_auth_aws_session_token]

```{applies_to}
stack: ga 9.3.0
```

> The AWS session token that can be optionally set when direct credentials([`auth.aws.access_key_id`](#_auth_aws_access_key_id) and [`auth.aws.secret_access_key`](#_auth_aws_secret_access_key)) are used.

Reviewer: ```suggestion
The AWS session token that can be optionally set when direct keys ([`auth.aws.access_key_id`](#_auth_aws_access_key_id) and [`auth.aws.secret_access_key`](#_auth_aws_secret_access_key)) are used.
```

### `auth.aws.shared_credential_file` [_auth_aws_shared_credential_file]

```{applies_to}
stack: ga 9.3.0
```

The path of the AWS shared credentials file.

::::{note}

> Direct keys ([`auth.aws.access_key_id`](#_auth_aws_access_key_id) and [`auth.aws.secret_access_key`](#_auth_aws_secret_access_key)) can not be set alongside with shared credentials file ([`auth.aws.shared_credential_file`](#_auth_aws_shared_credential_file)) if both are set, the direct keys takes precedence.

Reviewer: ```suggestion
Use either direct keys ([`auth.aws.access_key_id`](#_auth_aws_access_key_id) and [`auth.aws.secret_access_key`](#_auth_aws_secret_access_key)) or a shared credentials file ([`auth.aws.shared_credential_file`](#_auth_aws_shared_credential_file)). If both are set, the direct keys take precedence.
```
